core/version/src/db/dao.rs

```diff
@@ -0,0 +1,117 @@
+
+pub struct ReleaseDAO<'c> {
+    pool: &'c PoolType,
+}
+impl<'a> AsDao<'a> for ReleaseDAO<'a> {
+    fn as_dao(pool: &'a PoolType) -> Self {
+        Self { pool }
+    }
+}
+
+impl<'c> ReleaseDAO<'c> {
+    pub async fn save(&self, db_rel: DBRelease) -> anyhow::Result<Release> {
+        do_with_transaction(self.pool, move |conn| {
+            match get_release(conn, &db_rel.version)? {
+                Some(rel) => Ok(rel),
+                None => {
+                    diesel::insert_into(version_release)
+                        .values(&db_rel)
+                        .execute(conn)?;
+                    Ok(db_rel.into())
+                }
+            }
+        })
+        .await
+    }
+
+    pub async fn current_release(&self) -> anyhow::Result<Option<Release>> {
+        readonly_transaction(self.pool, move |conn| get_current_release(conn)).await
+    }
+
+    pub async fn pending_release(&self) -> anyhow::Result<Option<Release>> {
+        readonly_transaction(self.pool, move |conn| get_pending_release(conn)).await
+    }
+
+    pub async fn version(&self) -> anyhow::Result<VersionInfo> {
+        log::debug!("Getting Yagna version: current and pending from DB");
+        readonly_transaction(self.pool, move |conn| {
+            Ok(VersionInfo {
+                current: get_current_release(conn)?.ok_or(anyhow!("Can't get current release."))?,
+                pending: get_pending_release(conn)?,
+            })
+        })
+        .await
+    }
+
+    pub async fn skip_pending_release(&self) -> anyhow::Result<Option<Release>> {
+        log::debug!("Skipping latest pending Yagna release");
+        do_with_transaction(self.pool, move |conn| {
+            let mut pending_rel = match get_pending_release(conn)? {
+                Some(rel) => rel,
+                None => return Ok(None),
+            };
+            let num_updated = diesel::update(version_release.find(&pending_rel.version))
+                .set(release::seen.eq(true))
+                .execute(conn)?;
+            pending_rel.seen = true;
+            match num_updated {
+                0 => anyhow::bail!("Release not skipped: {}", pending_rel),
+                1 => Ok(Some(pending_rel)),
+                _ => anyhow::bail!("More than one release skipped: {}", pending_rel),
+            }
+        })
+        .await
+    }
+}
+
+fn get_current_release(conn: &ConnType) -> anyhow::Result<Option<Release>> {
+    get_release(conn, ya_compile_time_utils::semver_str())
+}
+
+fn get_release(conn: &ConnType, ver: &str) -> anyhow::Result<Option<Release>> {
+    Ok(version_release
+        .filter(release::version.eq(&ver))
+        .first::<DBRelease>(conn)
+        .optional()?
+        .map(|db_rel| db_rel.into()))
+}
+
+fn get_pending_release(conn: &ConnType) -> anyhow::Result<Option<Release>> {
+    let query = version_release
+        .filter(release::seen.eq(false))
+        .order((release::release_ts.desc(), release::version.desc()))
```

Comment:
I'm afraid that ordering by version won't work here. (`0.9.0' > '0.10.0`)

Reply:
good catch

Reply:
ok, now's that `release_ts` is unique I can remove the second column from `order by`